[WIP] Refer Screen

aliseyalvi opened the pull request:
### What does this PR?
Work In Progress on Refer Screen

### Issue number
#2182 

### Screenshots/Video
[WIP]

aliseyalvi commented:
@noumantahir need further instructions on this to process further.

- Added Basic Refer Screen UI with referred users list
- Implemented referrals API with converter and referral interface type defined

See the screenshot below. Currently using good-karma username for testing . I have no account with referrals.

![image](https://user-images.githubusercontent.com/48380998/154400767-ee7a0d96-16e5-4d1d-a6b5-79392ed2f564.png)

noumantahir commented on src/screens/referScreen/referScreen.tsx:
```diff
@@ -0,0 +1,98 @@
+  const _renderPointsEarned = () => {
+    return (
+      <View style={styles.pointsEarnedContainer}>
+        <Text style={styles.points}>1000</Text>
```
points earned depend on rewarded referrals, as suggested by @feruzm we can multiply rewarded number of users with 100 to show real rewarded points. We can also show estimated value of points. 

And for unrewarded ones, how about we show "Earning Pending" based on unrewarded items

noumantahir commented on src/screens/referScreen/referScreen.tsx:
```diff
@@ -0,0 +1,98 @@
+          onPress={() => console.log('pressed!')}
+        >
+          <View style={styles.mainButtonWrapper}>
+            <Text style={styles.unclaimedText}>Refer</Text>
```
use intl and all caps REFER.. also instead of + icon how about we use a share icon

noumantahir commented on src/screens/referScreen/referScreen.tsx:
```diff
@@ -0,0 +1,98 @@
+        isHasRightItem
+        isClickable
+        isBlackRightColor
+        rightText={item.isRewarded ? 'Rewarded' : 'Not Rewarded'}
```
instead of Rewarded Unrewarded text, may we can show blue double dollar icon and gray double dollar icon... @feruzm what do you think... as for show filling bar, apparently the api do not return enough information for that and making api calls for each user may not be a good approach. double dollar can have tooltip to show what it represent

Delegate HP button on right..

Check screenshot for reference, also added figma

<img width="321" alt="Screenshot 2022-02-17 at 9 56 00 PM" src="https://user-images.githubusercontent.com/6298342/154531477-11071abb-bacf-4200-966e-a01965ccbd12.png">

feruzm replied:
I like this approach, delegate is good to make new people feel welcomed. We will show tutor tooltip to give that hint and other details could be explained that way as well.

aliseyalvi commented:
@noumantahir please check the video below for latest update.

https://user-images.githubusercontent.com/48380998/154622736-95534bd2-2f0e-4af3-bcd0-6f4c09a2437b.mp4

noumantahir commented:
@aliseyalvi make sure to test the dark mode..

<img width="285" alt="Screenshot 2022-02-18 at 10 15 42 PM" src="https://user-images.githubusercontent.com/6298342/154730679-c41b6b50-0f1a-462b-a5eb-0e3a528e6472.png">

aliseyalvi commented:
@noumantahir check the update onwards https://github.com/ecency/ecency-mobile/pull/2188/commits/27675be7513d17b4b66abf5a7eda9a972ef5e46d.

- added referrals flatlist pagination.
- integrated referrals stats endpoint
- added flatlist refreshing
- fixed autofocus issue when navigating to delegation screen.

noumantahir reviewed (changes requested):
@aliseyalvi just one update required, open user profile on item tap.....other than that all good.. just added a few minor tweaks I added, do check them out. other than that I think we can merge...

noumantahir commented:
@feruzm please review demo video as well so we can remove the test username and replace with real one.

Pending Item: Opening user profile on item tap.

https://user-images.githubusercontent.com/6298342/154961004-9118c666-c072-4ad0-8468-ff23791f1026.mov
![Screenshot 2022-02-21 at 6 04 15 PM](https://user-images.githubusercontent.com/6298342/154961177-6142924c-c649-465e-b1dd-f0a3a7c3393c.png)

noumantahir commented:
wallet branch is now being built in app centre, beta should be available as build is complete

aliseyalvi commented:
@noumantahir 
- added navigate to user profile on list item press https://github.com/ecency/ecency-mobile/pull/2188/commits/f23fb3e8fd5c3c25b09e712e60ae43759f9c9e14

aliseyalvi commented on src/screens/transfer/screen/delegateScreen.js:
```diff
@@ -70,6 +70,14 @@ class DelegateScreen extends Component {
   }
 
   // Component Lifecycles
+  componentDidMount() {
```
https://github.com/ecency/ecency-mobile/pull/2188/commits/c41afeefa8f8eb1145b41831acbbab8f704308bb

noumantahir commented:
@feruzm seems all good from my side, will be great if you could also have a quick look...

feruzm commented:
@noumantahir looks good, can we change earned text to Earned Points to keep consistency on wording?

![85CD9E99-1820-4C7B-9933-9F2B697E80C9](https://user-images.githubusercontent.com/3015990/155007157-1106a711-2143-40d6-a127-bf578323cbfd.jpeg)

noumantahir commented:
@feruzm done
